libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -231,50 +232,89 @@ end
-    angles4[1] = 45 + angles4[1]
+    local ans_RB = body_forward_kinematics(endpoint_RB[1]+gait_pos_x[1],
+                                         endpoint_RB[2]+gait_pos_y[1], 
+                                         endpoint_RB[3]+gait_pos_z[1],
+                                         -FRAME_LEN*0.5, FRAME_WIDTH*0.5, 
+                                         gait_rot_z[1])
+    local angles_RB = leg_inverse_kinematics(endpoint_RB[1]+ans_RB[1]+gait_pos_x[1],
+                                             endpoint_RB[2]+ans_RB[2]+gait_pos_y[1], 
+                                             endpoint_RB[3]+ans_RB[3]+gait_pos_z[1])
+    angles_RB[1] = 45 + angles_RB[1]
+
+    local ans_RF = body_forward_kinematics(endpoint_RF[1]+gait_pos_x[2],
+                                         endpoint_RF[2]+gait_pos_y[2],
+                                         endpoint_RF[3]+gait_pos_z[2],
+                                         -FRAME_LEN*0.5, -FRAME_WIDTH*0.5,
+                                         gait_rot_z[2])
+    local angles_RF = leg_inverse_kinematics(endpoint_RF[1]-ans_RF[1]+gait_pos_x[2],
+                                             endpoint_RF[2]-ans_RF[2]-gait_pos_y[2], 
+                                             endpoint_RF[3]+ans_RF[3]+gait_pos_z[2])
+    angles_RF[1] = 135 + angles_RF[1]
+
+    local ans_LB = body_forward_kinematics(endpoint_LB[1]+gait_pos_x[3],
+                                         endpoint_LB[2]+gait_pos_y[3], 
+                                         endpoint_LB[3]+gait_pos_z[3], 
+                                         FRAME_LEN*0.5, FRAME_WIDTH*0.5,
+                                         gait_rot_z[3])
+    local angles_LB = leg_inverse_kinematics(endpoint_LB[1]+ans_LB[1]+gait_pos_x[3],
+                                             endpoint_LB[2]+ans_LB[2]+gait_pos_y[3], 
+                                             endpoint_LB[3]+ans_LB[3]+gait_pos_z[3])
+    angles_LB[1] = -45 + angles_LB[1]
+
+    local ans_LF = body_forward_kinematics(endpoint_LF[1]+gait_pos_x[4],
+                                         endpoint_LF[2]+gait_pos_y[4], 
+                                         endpoint_LF[3]+gait_pos_z[4], 
+                                         FRAME_LEN*0.5, -FRAME_WIDTH*0.5, 
+                                         gait_rot_z[4])
+    local angles_LF = leg_inverse_kinematics(endpoint_LF[1]-ans_LF[1]+gait_pos_x[4],
+                                             endpoint_LF[2]-ans_LF[2]-gait_pos_y[4], 
+                                             endpoint_LF[3]+ans_LF[3]+gait_pos_z[4])
+    angles_LF[1] = -135 + angles_LF[1]
     Gaitselect()
-    current = {angles1[1],angles1[2],angles1[3],angles2[1],angles2[2],angles2[3],angles3[1],angles3[2],angles3[3],angles4[1],angles4[2],angles4[3]}
+    current_angle = {angles_RF[1],angles_RF[2],angles_RF[3],
```

Comment:
`current_angle` appears to be an entry in the global table, it should probably be local to this function, and passed as an argument to `servo_estimate()` rather then being a lookup from the global state.